azurelinuxagent/common/osutil/openbsd.py

```diff
@@ -106,7 +106,7 @@ def chpasswd(self, username, password, crypt_id=6, salt_len=10):
         if self.is_sys_user(username):
             raise OSUtilError(("User {0} is a system user. "
                                "Will not set passwd.").format(username))
-        output = self._run_command_raising_OSUtilError(['echo', '-n', '{0}|encrypt'.format(password)],
```

Comment:
echo -n basically just does not append \n at the end of the string. They were using echo to redirect password to the encrypt command. Passing it through Popen now